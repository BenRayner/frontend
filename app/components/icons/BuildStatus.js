import React from 'react';

export default class BuildStatus extends React.Component {
  static propTypes = {
    status: React.PropTypes.oneOf(["pending", "scheduled", "running", "passed", "paused", "failed", "canceled", "skipped"]).isRequired,
    size: React.PropTypes.oneOf(["regular","small"]).isRequired,
    className: React.PropTypes.string
  };

  static defaultProps = {
    size: 'regular'
  };

  render() {
    return (this.props.size == 'regular') ? this._renderRegular() : this._renderSmall();
  }

  _renderRegular() {
    return (
      <svg width="32px" height="32px" viewBox="0 0 32 32" className={this.props.className}>
        <g transform="translate(-219.000000, -19.000000)" stroke="#F83F23" strokeWidth="2">
          <g transform="translate(220.000000, 20.000000)">
            <ellipse fill="none" cx="15" cy="15" rx="15" ry="15"></ellipse>
            <g transform="translate(9.000000, 9.000000)">
              <path d="M0.600275489,0.600275489 L11.3997245,11.3997245"></path>
              <path d="M11.3997245,0.600275489 L0.600275489,11.3997245"></path>
            </g>
          </g>
        </g>
      </svg>
    )
  }

  _renderSmall() {
    return (
      <svg width="20px" height="20px" viewBox="0 0 20 20" className={this.props.className}>
      </svg>
    )
  }
}
