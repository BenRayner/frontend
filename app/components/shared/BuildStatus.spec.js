/* global describe, it, expect */
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import BuildStatus from './BuildStatus';

const BUILD_STATES = [
  'started',
  'failed',
  'passed',
  'blocked',
  'canceled',
  'canceling',
  'scheduled',
  'skipped',
  'not_run'
];

describe('BuildStatus', () => {
  BUILD_STATES.forEach((state) => {
    it(`shows correct build status for the \`${state}\` state`, () => {
      const build = {
        state,
        canceledAt: new Date(1475643467000),
        createdAt: new Date(1475643466920),
        finishedAt: new Date(1475643467000)
      };

      const component = ReactTestRenderer.create(
        <BuildStatus build={build} updateFrequency={0} />
      );

      const tree = component.toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
