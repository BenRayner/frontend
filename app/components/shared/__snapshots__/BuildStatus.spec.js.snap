exports[`BuildStatus shows correct build status for the \`blocked\` state 1`] = `
<span>
  Blocked
   
  <time
    dateTime={2016-10-05T04:57:47.000Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`canceled\` state 1`] = `
<span>
  Canceled
   
  <time
    dateTime={2016-10-05T04:57:47.000Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`canceling\` state 1`] = `
<span>
  Canceled
   
  <time
    dateTime={2016-10-05T04:57:47.000Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`failed\` state 1`] = `
<span>
  Finished
   
  <time
    dateTime={2016-10-05T04:57:47.000Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`not_run\` state 1`] = `
<span>
  Created
   
  <time
    dateTime={2016-10-05T04:57:46.920Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`passed\` state 1`] = `
<span>
  Finished
   
  <time
    dateTime={2016-10-05T04:57:47.000Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`scheduled\` state 1`] = `
<span>
  Created
   
  <time
    dateTime={2016-10-05T04:57:46.920Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`skipped\` state 1`] = `
<span>
  Skipped
   
  <time
    dateTime={2016-10-05T04:57:46.920Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;

exports[`BuildStatus shows correct build status for the \`started\` state 1`] = `
<span>
  Created
   
  <time
    dateTime={2016-10-05T04:57:46.920Z}
    title="Wednesday, October 5, 2016 3:57 PM">
    Wednesday at 3:57 PM
  </time>
</span>
`;
